### PR TITLE
Make benchmark runner capable of testing multiple branches

### LINKED ISSUE #2967: Make benchmark runner able to run tests against both "master" and current branch
💭 We can publish a browserified build of `master`'s benchmarks to s3

### PR DESCRIPTION
**Pull request by lucaswoj**
We require all substantial PRs to run the benchmark suite on their branch and `master`. This is important because the benchmark scores are a function of the system on which they are running. The deltas between the branch under test and `master` is the strongest signal we can measure. 

Switching between branches to do this test can be annoying, especially if there are a bunch of `npm link`ed modules. This PR extends the benchmark runner to be capable of running both the branch under test and `master`'s benchmarks at once. 

`master`'s benchmark bundle is uploaded to s3 by CircleCI. As a bonus, this tests our deployment infrastructure more regularly. 

resolves #2967

cc @mourner @jfirebaugh @mollymerp @mapsam 

## Checklist

 - [x] briefly describe the changes in this PR
 - [ ] get a PR review :+1: / :-1:

### CODE REVIEW

**Comment by lucaswoj** [comment]
Can I get some 👀 on this today? Most of the code changes are pretty boring 😄 @jfirebaugh @ansis @mourner

**Comment by jfirebaugh** [comment]
I'm not too familiar with benchmark code, and I haven't reviewed this in depth. It seems like a welcome addition though. If you're happy I think you should merge.

**Comment by ansis** [comment]
This looks very useful

**Comment by mourner** [comment]
@lucaswoj after commiting my change in a branch and then `npm start`, the benchmark only runs the master branch and not my local branch. Is there anything I'm missing or is this a bug?

image

**Comment by lucaswoj** [comment]
Looks like you fixed the bug ☝️ in #3173. Thanks 😄.

File: ci/dependencies.sh [hunk 1]
```diff
@@ -12,3 +12,7 @@ if [ "$CIRCLE_BRANCH" == "master" ]; then
 else
     source ./nvm/nvm.sh && nvm use ${NODE_VERSION} && npm install
 fi
+
+if [ "$CIRCLE_BRANCH" == "master"] || [ -z "$CIRCLE_TAG" ]; then
+    pip install --user --upgrade awscli
+fi
```
Comment (mourner) [inline ci/dependencies.sh]: @lucaswoj Circle complains with `./ci/dependencies.sh: line 16: [: missing `]'` on this part, and then proceeds to install awscli despite not being on master or having a tag. Not that it takes much time, but would be nice to fix.

Comment (lucaswoj) [inline ci/dependencies.sh]: Thanks for the catch! I think I know the problem.

**Pull request merged**